src/components/App.js

```diff
@@ -1,15 +1,161 @@
-import React from 'react';
+import React, { Component } from 'react';
+import { BrowserRouter, Route } from 'react-router-dom';
+import { getWeatherInformation, toCelcius, toFahrenheit } from '../utils/weather';
+import getLocation from '../utils/location';
+
+import UVIndex from './UVIndex';
+import Forecast from './Forecast';
+import Nav from './Nav';
+import UnitToggle from './UnitToggle';
 import Weather from './Weather';
 
-const App = () => (
-  <div>
-    <h1>React Boilerplate with Sass</h1>
-    <img
-      src="https://upload.wikimedia.org/wikipedia/commons/a/a7/React-icon.svg"
-      alt="react-logo"
-    />
-    <Weather />
-  </div>
-);
+class App extends Component {
+  constructor() {
+    super();
+    this.state = {
```

Comment:
** Another PR **
I think a lot of your state is not shared between components. Place state as close to where it is needed as possible. Similar to my philosophy about "utility" functions etc.